include/git2/apply.h

```diff
@@ -72,6 +81,9 @@ typedef struct {
 
 	/** Payload passed to both delta_cb & hunk_cb. */
 	void *payload;
+
+	/** Bitmask of git_apply_flags_t */
+	git_apply_flags_t flags;
```

Comment:
I think this field should rather be an unsigned integer as it will not be an enum itself, but rather an OR'd value of different enums.

Reply:
Also based on existing code throughout libgit2, but I was initially going to go for an unsigned int. Unsure of how to proceed

Reply:
Yeah, I realize now that we've been inconsistent here. #5242 has fixed some of these locations, so this here should also use an unsigned integer